_source/_posts/2024-02-20-net-scim.md

```diff
@@ -0,0 +1,1909 @@
+```json
+"ConnectionStrings": {
+    "DefaultConnection": "Data Source=scim-dev.db;"
+}
+```
+
+This command above creates a db file in the project folder named `scim-db.dev`. After adding this property, the JSON file should look like this:
+```json
+{
+  "ConnectionStrings": {
+    "DefaultConnection": "Data Source=scim-dev.db;"
+  },
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft.AspNetCore": "Warning"
+    }
+  },
+  "AllowedHosts": "*"
+}
+```
+Now we can wire it all up in `Program.cs` for DI using `AddDbContextPool` middleware
+- At the top of the file, add dependencies with `using` statements
+    ```c#
+    using Microsoft.EntityFrameworkCore;
+    using okta_scim_server_dotnet;
+    ```
+- After `builder.Services.AddSwaggerGen();` add the following code:
+    ```c#
+    builder.Services.AddDbContextPool<ScimDbContext>(
+        options => options.UseSqlite(builder.Configuration.GetConnectionString("DefaultConnection"))
+    );
+    ```
+The `Program.cs` file should now look like this:
+
+```c#
+using Microsoft.EntityFrameworkCore;
+using okta_scim_server_dotnet; 
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+builder.Services.AddDbContextPool<ScimDbContext>(
+    options => options.UseSqlite(builder.Configuration.GetConnectionString("DefaultConnection"))
+);
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
+{
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.UseHttpsRedirection();
+
+var summaries = new[]
+{
+    "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
+};
+
+app.MapGet("/weatherforecast", () =>
+{
+    var forecast =  Enumerable.Range(1, 5).Select(index =>
+        new WeatherForecast
+        (
+            DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
+            Random.Shared.Next(-20, 55),
+            summaries[Random.Shared.Next(summaries.Length)]
+        ))
+        .ToArray();
+    return forecast;
+})
+.WithName("GetWeatherForecast")
+.WithOpenApi();
+
+app.Run();
+
+record WeatherForecast(DateOnly Date, int TemperatureC, string? Summary)
+{
+    public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);
+}
+```
+
+We can then migrate the coded database models and data into the actual database.
+- Prepare db migration by running `dotnet ef migrations add InitialScimDb`
+- Apply these changes to db by running `dotnet ef database update`. This command would have created a file named `scim-dev.db` in your project folder.
+- *Optional:* Test db creation using the command line tool, 
+    - You should have sqllite3 client installed. (I had this OOB in Mac OS)
```

Comment:
Typo. SHould be sqlite3.

Also, we shouldn't make this assumption. Link how to install if it doesn't exist.